go/chat/convloader.go

```diff
@@ -156,27 +159,30 @@ func (b *BackgroundConvLoader) removeActiveLoadLocked(key string) {
 	delete(b.activeLoads, key)
 }
 
-func (b *BackgroundConvLoader) monitorAppState() {
+func (b *BackgroundConvLoader) monitorAppState() error {
 	ctx := context.Background()
-	suspended := false
 	b.Debug(ctx, "monitorAppState: starting up")
+
+	suspended := false
 	state := keybase1.MobileAppState_FOREGROUND
 	for {
-		state = <-b.G().MobileAppState.NextUpdate(&state)
-		switch state {
-		case keybase1.MobileAppState_FOREGROUND, keybase1.MobileAppState_BACKGROUNDACTIVE:
-			b.Debug(ctx, "monitorAppState: active state: %v", state)
-			// Only resume if we had suspended earlier (frontend can spam us with these)
-			if suspended {
-				b.Debug(ctx, "monitorAppState: resuming load thread")
-				b.Resume(ctx)
-				suspended = false
-			}
-		case keybase1.MobileAppState_BACKGROUND:
-			b.Debug(ctx, "monitorAppState: backgrounded, suspending load thread")
-			if !suspended {
-				b.Suspend(ctx)
-				suspended = true
+		select {
+		case state = <-b.G().MobileAppState.NextUpdate(&state):
```

Comment:
I'm not sure this is how we want to do this. Often times I am on good LTE and would appreciate this to run. I think a better strategy would be controlling how many jobs get queued into this on LTE, probably in `Syncer` and `InboxSource`.

Reply:
i'm often on throttled data so the connection type is lte but it's limited by my provider, was hoping to cut down bandwidth in this case. we could also plumb through the connection type and limit on LTE and stop completely on 4g or something

Reply:
In `Syncer` we load conversations after they are synced, and you will likely be tapping them soon anyway. My proposal was something smarter on that end, where it doesn't queue up more than like 3 convs and they all have to be imp or small teams convs.